src/studio/src/designer/frontend/packages/schema-editor/src/components/SchemaInspector.tsx

```diff
@@ -154,6 +160,15 @@ const SchemaInspector = ((props: ISchemaInspectorProps) => {
       setIsRequired(parentItem?.required?.includes(selectedItem?.displayName) ?? false);
       if (selectedItem.$ref !== undefined || selectedItem.items?.$ref !== undefined) {
         setObjectKind('reference');
+      } else if (selectedItem.allOf || selectedItem.anyOf || selectedItem.oneOf) {
```

Comment:
Instead of having `allOf`, `anyOf` etc as props on the node, why not have something like:
```js
{
  'combination': [ ...(data) ],
  'combinationKind': 'allOf' | 'anyOf' | 'oneOf',
}
```
Then this block would likely be a little less complex?

Reply:
I mean, we don't actually need to store the combination inside a prop with the name `allOf`, `anyOf`, etc..

Reply:
Really liked this suggestion. Refactored.